filters/tee/teeloopback.go

```diff
@@ -0,0 +1,65 @@
+package tee
+
+import (
+	"context"
+	log "github.com/sirupsen/logrus"
+	"github.com/zalando/skipper/filters"
+	teePredicate "github.com/zalando/skipper/predicates/tee"
+	"net/url"
+)
+
+const FilterName = "teeLoopback"
+
+type teeLoopbackSpec struct{}
+type teeLoopbackFilter struct {
+	teeKey string
+}
+
+func (t *teeLoopbackSpec) Name() string {
+	return FilterName
+}
+
+func (t *teeLoopbackSpec) CreateFilter(args []interface{}) (filters.Filter, error) {
+	if len(args) < 1 {
+		return nil, filters.ErrInvalidFilterParameters
+	}
+	teeKey, ok := args[0].(string)
+	if !ok || teeKey == "" {
+		return nil, filters.ErrInvalidFilterParameters
+	}
+	return &teeLoopbackFilter{
+		teeKey,
+	}, nil
+}
+
+func NewTeeLoopback() filters.Spec {
+	return &teeLoopbackSpec{}
+}
+
+func (f *teeLoopbackFilter) Request(ctx filters.FilterContext) {
+	origRequest := ctx.Request()
+	// prevent the loopback to be executed indefinitely
```

Comment:
this protection already exists with a configurable limit in the proxy loopback implementation, with a configurable counter. I think we should reuse it.

Reply:
I have tested this works properly I need to figure out how to write a test for this.